fe/lib/components/LocationSelectMap/LocationSelectMap.tsx

```diff
@@ -0,0 +1,191 @@
+    mobile: number;
+  };
+}
+
+export const LocationSelectMap = ({
+  debounceDelay = 250,
+  schemeId,
+  onLocationSelected,
+  initialLocation,
+  client,
+  mapHeight = {
+    wide: DEFAULT_MAP_HEIGHT,
+    desktop: DEFAULT_MAP_HEIGHT,
+    tablet: 200,
+    mobile: 200,
+  },
+}: LocationSelectMapProps) => {
+  const [latLong, setLatLong] = useState(initialLocation);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const responsiveValue = useResponsiveValue();
+  const color = useColor();
+  const [debouncedLatLong] = useDebounce(latLong, debounceDelay);
+
+  const { data, previousData } = useQuery<NearbyLocationsQuery>(
+    NEAREST_LOCATIONS,
+    {
+      variables: {
+        geoLocation: {
+          latitude: debouncedLatLong[0],
+          longitude: debouncedLatLong[1],
+        },
+        schemeId,
+      },
+      client,
+    },
+  );
+
+  const nearestLocations =
+    data?.nearestLocations ?? previousData?.nearestLocations;
+
+  const isDesktopOrAbove = responsiveValue({
+    mobile: false,
+    desktop: true,
+  });
+
+  const height = responsiveValue(mapHeight) ?? DEFAULT_MAP_HEIGHT;
+
+  const NearestLocationsCards = () => (
+    <Box borderRadius="standard" background="surface">
+      <Stack dividers space="none">
+        <Box
+          paddingX="medium"
+          paddingY={{ desktop: 'xsmall', tablet: 'small', mobile: 'small' }}
+        >
+          <Columns space="small" alignY="center">
+            <Column>
+              <Text weight="strong" size="small">
+                {`Locations near ${formatPoint(latLong)}`}
+              </Text>
+            </Column>
+            {isDesktopOrAbove ? (
+              <Column width="content">
+                <ButtonIcon
+                  icon={<IconClear />}
+                  tone="secondary"
+                  id="clear-nearest-suggestions-button"
+                  label="Clear Suggestions"
+                  onClick={() => setShowSuggestions(false)}
+                />
+              </Column>
+            ) : null}
+          </Columns>
+        </Box>
+        {nearestLocations?.map((location) => (
+          <Box
+            className={styles.card}
+            cursor="pointer"
+            padding="medium"
+            key={location.id.value}
+            onClick={() => onLocationSelected(location)}
+          >
+            <Stack space="small">
+              <Text size="small">{location.contextualName}</Text>
+              <InlineCode>{location.id.value}</InlineCode>
+            </Stack>
+          </Box>
+        ))}
+      </Stack>
+    </Box>
+  );
+
+  const NoLocationsNotice = () => (
+    <Alert
+      tone="info"
+      onClose={() => setShowSuggestions(false)}
+      closeLabel="Close"
+    >
+      <Text>No locations found.</Text>
+    </Alert>
+  );
+
+  const hasNearestLocations = nearestLocations && nearestLocations.length > 0;
+  const suggestionsOffset: [number, number] = [-40, 130];
+
+  return (
+    <Box>
```

Comment:
Would this be more descriptive if it were a `Stack space="none"`?

Reply:
Yep I think it is 👍